### PR TITLE
rework vis data gathering sections

### LINKED ISSUE #393: Fix visualization section being outdated.

### PR DESCRIPTION
**Pull request by dyc3**
closes #393

> [!CAUTION]
> DO NOT merge yourself, this will be auto merged upon approval

### CODE REVIEW

**Comment by dyc3** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dyc3/senior-design/411?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#411** <a href="https://app.graphite.dev/github/pr/dyc3/senior-design/411?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#410** <a href="https://app.graphite.dev/github/pr/dyc3/senior-design/410?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#409** <a href="https://app.graphite.dev/github/pr/dyc3/senior-design/409?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#408** <a href="https://app.graphite.dev/github/pr/dyc3/senior-design/408?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#407** <a href="https://app.graphite.dev/github/pr/dyc3/senior-design/407?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#406** <a href="https://app.graphite.dev/github/pr/dyc3/senior-design/406?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#405** <a href="https://app.graphite.dev/github/pr/dyc3/senior-design/405?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#404** <a href="https://app.graphite.dev/github/pr/dyc3/senior-design/404?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#402** <a href="https://app.graphite.dev/github/pr/dyc3/senior-design/402?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#403](https://github.com/dyc3/senior-design/pull/403) <a href="https://app.graphite.dev/github/pr/dyc3/senior-design/403?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#401** <a href="https://app.graphite.dev/github/pr/dyc3/senior-design/401?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#399** <a href="https://app.graphite.dev/github/pr/dyc3/senior-design/399?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#398** <a href="https://app.graphite.dev/github/pr/dyc3/senior-design/398?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#397** <a href="https://app.graphite.dev/github/pr/dyc3/senior-design/397?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#396** <a href="https://app.graphite.dev/github/pr/dyc3/senior-design/396?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#400** <a href="https://app.graphite.dev/github/pr/dyc3/senior-design/400?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @dyc3 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: visualization-design.typ [hunk 1]
```diff
@@ -110,20 +110,18 @@ client = {
...
 
-Grafana supports querying Prometheus, and there is documentation linked below on a quick start for creating a new data source. Given the Balancer has already integrated Prometheus for metrics, this is the preferred method for gathering data. There is no mention on compatibility with D3.js, but assuming there are no issues integrating D3.js into a Grafana panel, this should not be a problem. #cite(<grafana-prometheus-visualization>)
+Grafana supports querying Prometheus, and supports custom data sources through it's plugin system. Given the Balancer has already integrated Prometheus for metrics, this is the preferred method for gathering data. #cite(<grafana-prometheus-visualization>) Grafana is a tool primarily meant for time series data, and no current data source plugins support receiving the type of information needed. To add support, a custom data source must be created.
```
Comment (Victor-M-Giraldo) [inline visualization-design.typ:113]: Change it's to its.

File: visualization-design.typ [hunk 2]
```diff
@@ -110,20 +110,18 @@ client = {
...
 
 === Balancer Discovery
 
-While the visualization is running: Multiple instances of the Balancer can be active simultaneously, new instances can become active, and instances can go offline. The addresses of these Balancers are not known at runtime, so a discovery process similar to @Chapter::ServiceDiscovery must run to collect data from the discovered Balancers.
-
-In order to achieve this, a new rust crate will be created to handle this discovery process. Implementation will likely be similar to @Chapter::ServiceDiscovery. A port will be opened to listen for active instances of the Balancer. When a connection or connections are found, the Balancer discoverer clones the Balancer(s) and connects.
+While the visualization is running, multiple instances of the Balancer can be active simultaneously, new instances can become active, and instances can go offline. The network addresses of these Balancers are not known at runtime, so the Collector must perform service discovery (as described in @Chapter::ServiceDiscovery) must run as a prerequisite in order to collect data from the Balancers.
```
Comment (Victor-M-Giraldo) [inline visualization-design.typ:119]:
```suggestion
While the visualization is running, multiple instances of the Balancer can be active simultaneously, new instances can become active, and instances can go offline. The network addresses of these Balancers are not known at runtime, so the Collector must perform service discovery (as described in @Chapter::ServiceDiscovery). It must run as a prerequisite in order to collect data from the Balancers.
```

**Review by Victor-M-Giraldo — CHANGES_REQUESTED** [review]

**Review by Victor-M-Giraldo — APPROVED** [review]

**Comment by dyc3** [comment]
### Merge activity

* **Apr 17, 4:14 PM EDT**: @dyc3 merged this pull request with [Graphite](https://app.graphite.dev/github/pr/dyc3/senior-design/411).

**Pull request merged**